addons/File_Sender/src/presenter.js

```diff
@@ -0,0 +1,699 @@
+                    }
+                    presenter.ancestorsData = undefined;
+                }
+            }
+
+            if (isPopup || presenter.isPreview) {
+                popupLeft = presentationPosition.left;
+                topPosition = parseInt((availableHeight - dialogHeight) / 2, 10);
+            }
+            else {
+                topPosition = parseInt(( windowHeight - dialogHeight) / 2, 10) ;
+            }
+
+            var presentationHorizontalOffset = parseInt((presentationWidth - dialogWidth) * scaleInfo.scaleY / 2, 10);
+            var leftPosition = presentationPosition.left + presentationHorizontalOffset;
+
+            // adjust top position if Player was embedded in iframe (i.e. EverTeach)
+            if (window !== window.top) {
+                var iframe = window.parent.document.getElementsByTagName('iframe');
+                var iframeDialogHeight = parseInt($dialog.height(), 10);
+                iframeDialogHeight += DOMOperationsUtils.calculateOuterDistances(DOMOperationsUtils.getOuterDimensions($dialog)).vertical;
+
+                if (topPosition < 0) {
+                    topPosition = 0;
+                } else if (topPosition > $(window).height() - iframeDialogHeight) {
+                    topPosition = $(window).height() - iframeDialogHeight;
+                }
+            }
+
+
+            if ($(window).scrollTop() > popupTop && isPopup) {
+                topPosition += ($(window).scrollTop() - popupTop);
+            }
+
+            $dialog.css({
+                left: (leftPosition - popupLeft) + 'px',
+                top: (topPosition + scrollTop + previewFrame) + 'px'
+            });
+
+            $dialog.find('.ui-dialog-content').css({
+                color: 'black'
+            });
+
+            if(isPopup || presenter.isPreview) {
+                // For Preview and Popup dialog is moved to appropriate page
+                var $overlay = $(".ui-widget-overlay");
+                $(presenter.$view.closest(".ui-widget-overlay")).remove();
+                if (isPreview) {
+                    $(".ic_page_panel").children(".ic_page").children().last().after($overlay);
+                }
+                else {
+                    $dialog.before($overlay);
+                }
+            }
+        }catch(e){console.log(e)}
+    };
+
+    presenter.closeDialogEventHandler = function() {
+        // due to the inability to close the dialog, when any video is under close button
+        try{
+            presenter.dialog.css("maxHeight", "none");
+
+            if (presenter.ancestorsData !== undefined) {
+                var ancestorData;
+                for (i=0; i<presenter.ancestorsData.length; i++)
+                {
+                    ancestorData = presenter.ancestorsData[i];
+                    $(ancestorData.wnd).scrollTop(ancestorData.offset);
+                }
+                presenter.ancestorsData = undefined;
+            }
+        }catch(e){}
```

Comment:
Zgodnie ze sztuką dobrze byłoby dać info o jakimś błędzie w catch - chociaż jako console.log, jeżeli to nic istotnego i spodziewamy się faila. W przyszłości może pomóc w debugowaniu.